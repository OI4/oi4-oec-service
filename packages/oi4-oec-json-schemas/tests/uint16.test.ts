import { matchersWithOptions } from 'jest-json-schema'
import schema from '../schemas/dataTypes/uint16.schema.json'

expect.extend(
  matchersWithOptions({
    verbose: true,
  })
)

describe('uint16 schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect(0).toMatchSchema(schema)
    expect(1).toMatchSchema(schema)
    expect(1.0).toMatchSchema(schema)
    expect(65534).toMatchSchema(schema)
    expect(65535).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    expect(-1).not.toMatchSchema(schema)
    expect(1.5).not.toMatchSchema(schema)
    expect(65536).not.toMatchSchema(schema)

    expect(() => {
      expect(-1).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('-1')
    expect(() => {
      expect(1.5).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('1.5')
    expect(() => {
      expect(65536).toMatchSchema(schema)
    }).toThrowErrorMatchingSnapshot('65536')
  })
})
