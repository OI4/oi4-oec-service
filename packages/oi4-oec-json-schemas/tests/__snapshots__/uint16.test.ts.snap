// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uint16 schema test invalid values: -1 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be >= 0
    Path:     dataTypes/uint16.schema.json#/minimum
"
`;

exports[`uint16 schema test invalid values: 1.5 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be integer
    Path:     dataTypes/uint16.schema.json#/type
"
`;

exports[`uint16 schema test invalid values: 65536 1`] = `
"expect(received).toMatchSchema(schema)

received
  must be <= 65535
    Path:     dataTypes/uint16.schema.json#/maximum
"
`;
